FlowCrypt/Controllers/Setup/SetupCreatePassphraseViewController.swift

```diff
@@ -188,6 +195,48 @@ extension SetupGenerateKeyViewController {
         }
     }
 
+    private func setupAccountWithExistingKeys(with passPhrase: String) {
+        Promise { [weak self] in
+            guard let self = self else { return }
+            self.showSpinner()
+
+            try awaitPromise(self.validateAndConfirmNewPassPhraseOrReject(passPhrase: passPhrase))
+
+            try self.keys.forEach { key in
+                try key.keyDetails.forEach { keyDetail in
+                    guard let privateKey = keyDetail.private else {
+                        throw CreatePassphraseWithExistingKeyError.noPrivateKey
+                    }
+                    let encryptedPrv = try self.core.encryptKey(
+                        armoredPrv: privateKey,
+                        passphrase: passPhrase
+                    )
+                    let parsedKey = try self.core.parseKeys(armoredOrBinary: encryptedPrv.encryptedKey.data())
+                    parsedKey.keyDetails.forEach { keyDetails in
+                        let passPhrase = PassPhrase(value: passPhrase, fingerprints: keyDetails.fingerprints)
+                        self.passPhraseService.savePassPhrase(with: passPhrase, inStorage: self.shouldSaveLocally)
+                    }
+                    try awaitPromise(self.backupService.backupToInbox(keys: parsedKey.keyDetails, for: self.user))
```

Comment:
Don't do this in this flow. The keys are already stored on EKM, no need to back them up anywhere else. This would be a severe vulnerability if merged in, easily enough to lose a customer if I didn't catch it in the PR review.

It shows a lack of clarity or awareness in our business use-case. I recommend to take more interest in the business side of things, understand not only what we do but why we do things in a certain way, study the documentation, and use the app itself a lot more, to educate yourself about our products.

Only with full understanding of what our company does and why, and how all of our products fit together, can you be a truly effective contributor.